### DISCUSSION TITLE
Component <unknown> took a long time for an operation (0.13 s)

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#107] [General]
How can I diagnose this? Any suggestions? I am using the lastest Dev code as of 7/25. I did not see this message before.
[14:29:58][W][component:204]: Component <unknown> took a long time for an operation (0.13 s).
[14:29:58][W][component:205]: Components should block for at most 20-30ms.
Thanks,

**Comment by Dilbert66** [comment]
Ignore it. That's just a newish warning from esphome. Nothing is broken.  It just doesnt like the fact that the custom component is taking a while to execute but nothing is impacted.

**Reply by AlexandreUSA** [reply]
Thanks. I just updated another eps board and it shows a similar error. I should stop updating multiple things at the same time...

Do you know why for the VistaAlarm, the compenent is <unknown> vs my other board that has the name of the sensor?

**Reply by Dilbert66** [reply]
I don't understand what you mean?

**Reply by AlexandreUSA** [reply]
My bad, some of the text got deleted...  I have the error in vistaalarm for an "unkown component" while for other ESP project, I see the name of the compoenent. I was wondering if I missed something in my configuration.

**Reply by Dilbert66** [reply]
if the component configuration has an ID line , it will be able to identify it.

**Reply by AlexandreUSA** [reply]
Thanks

**Comment by AlexandreUSA** [comment]
Closing the thread. Thanks again.
